### ISSUE TITLE
Allow readme.md and index.md files 

### ISSUE THREAD

**Issue by Sarabadu** [mkdocs/mkdocs#3110]
I have a use case having a `README.md` and an `index.md` file on my docs folder.

- The readme files content is repository specific and is the documentation that we want to show on github. 
- The index file is the project documentation homepage.

I tried to to define this on my `mkdocs.yml`: 
```yml
nav:
  - Introduction: index.md
  - Other: "other/index.md"
  - Repo Readme: "README.md"
```

I get that readme file is not considered as index if a `index.md` file is present, but why not allow still access to that file if we want to add that reference?

UPDATE
current vs expected behaviour
 
Current:
Add a file index.md to docs. A file index.html gets created in the site. The nav link leads there.
Add a file README.md to docs. A file index.html gets created in the site. The nav link leads there.
Add a file index.md to docs. Add a file README.md to docs. A file index.html gets created in the site. using the `index.md` file. The nav link leads there. the README.md file is ignored completely showing a warning about it

Expected
Add a file index.md to docs. A file index.html gets created in the site. The nav link leads there. (same)
Add a file README.md to docs. A file index.html gets created in the site. The nav link leads there. (same)
Add a file index.md to docs. Add a file README.md to docs. A file index.html gets created in the site. using the `index.md` file. The nav link leads there. a Readme.html file is created using the README.md file

**Comment by johnnovak** [comment]
Any movement on this? I wanted to do the exact same thing today, basically some discouraging in the README.md so people won't make changes to the GitHub Pages rendered content all the time instead of changing the sources...

**Comment by oprypin** [comment]
What is the "exact same thing" that you wanted to do??

This issue is incomplete or invalid. It doesn't specify what *should* happen.

Now:
* Add a file `index.md` to `docs`. A file `index.html` gets created in the `site`. The nav link leads there.
* Add a file `README.md` to `docs`. A file `index.html` gets created in the `site`. The nav link leads there.
----
* Add a file `index.md` to `docs`. Add a file `README.md` to `docs`. **WHAT** should happen? Two files with the same name created in quantum superposition that is resolved depending on which of two links the user clicks?

**Comment by johnnovak** [comment]
I thought it was obvious, but clearly you're not getting the point, so let me try again 😄 

MkDocs is often used to generate a website that then lives in a GitHub Pages repository. It would be very useful to be able to put a `README.md` file into the generated repo as part of the normal `mkdocs build` or `mkdocs deploy` process. That way when someone goes to the GitHub Pages repository page on GitHub, they would see the contents of the `README.md` file, which would contain instructions on the proper documentation process, pointing them to the MkDocs sources instead that possibly live in a different repo, and the instructions would also discourage them from raising PRs directly against the generated HTML content, which happens all the time, believe it or not...

In my case, I have a `README.md` and an `index.md` in my MkDocs sources at the root level. I want `index.html` to be generated from `index.md`, and I want `README.md` **only copied to the destination root folder**, and that's it! Simple!

So this is my source folder:

<img width="643" alt="image" src="https://user-images.githubusercontent.com/698770/217677360-835fcf4e-0f40-406d-bc5b-4442409c442d.png">

And this is my config:

<img width="573" alt="image" src="https://user-images.githubusercontent.com/698770/217677427-ab196b6b-b5e2-4d1d-958d-47afc4139bac.png">

If you search for it, a lot of people want this exact same feature for all sorts of documentation generators, not just MkDocs, because it's useful. So in this case, it would be a reasonable intelligent default for MkDocs to:

- Detect the ambiguity that you have correctly pointed out — we have both `index.md` and `README.md` which seem to server the same "role"
- But because `index.md` is referenced in the `nav` section, but `README.md` isn't, and `README.md` is a "known" special filename used by GitHub (and I imagine other services too), an intelligent resolution would be to just copy `README.md` over unaltered. The same way you're copying some other random files (e.g. image files) unaltered to the destination by default.
- Even if you don't want to do this intelligent conflict resolution, at least there should be an option to specify an "ignore list" of Markdown files that will be copied over unaltered, just as other resource files (e.g. images, audio files, etc.)

Hope that makes it clear now!

**Comment by Sarabadu** [comment]
@oprypin 
> Add a file index.md to docs. Add a file README.md to docs. WHAT should happen? Two files with the same name created in quantum superposition that is resolved depending on which of two links the user clicks?

What I expected to happen on my example is having the `index.html` file and a `Readme.html` file so I can still use both if I want to.

**Comment by oprypin** [comment]
* With #3224 you will be able to ignore README.md and just render index.md (without triggering a warning), or vice-versa.

* There is no plan to allow rendering README.md as README.html. A plugin or hook could do that if really desired. Or if there are further arguments provided, this could be revisited.
